src/libraries/System.Runtime.Serialization.Schema/src/System/Runtime/Serialization/Schema/ImportOptions.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.CodeDom;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+
+namespace System.Runtime.Serialization
+{
+    /// <summary>
+    /// Represents the options that can be set on an <see cref="XsdDataContractImporter"/>.
+    /// </summary>
+    /// <remarks>
+    /// The <see cref="XsdDataContractImporter"/> is used to generate code from XML schema using the .NET CodeDOM. To generate an XML schema from an assembly, use the <see cref="XsdDataContractExporter"/>.
+    /// </remarks>
+    public class ImportOptions
+    {
+        private ICollection<Type>? _referencedTypes;
+        private ICollection<Type>? _referencedCollectionTypes;
+        private IDictionary<string, string>? _namespaces;
+
+        /// <summary>
+        /// Gets or sets a <see cref="CodeDomProvider"/> instance that provides the means to check whether particular options for a target language are supported.
+        /// </summary>
+        public CodeDomProvider? CodeProvider { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that specifies whether types in generated code should implement the <see cref="System.ComponentModel.INotifyPropertyChanged"/> interface.
+        /// </summary>
+        public bool EnableDataBinding { get; set; }
+
+        /// <summary>
+        /// Gets or sets a data contract surrogate provider that can be used to modify the code generated during an import operation.
+        /// </summary>
+        /// <remarks>
+        /// The interface type for this option is ISerializationSurrogateProvider, but to take full advantage of the imported code modification
+        /// abilities, using an <see cref="ISerializationSurrogateProvider2"/>ISerializationSurrogateProvider2 that also implements <see cref="ISerializationCodeDomSurrogateProvider"/> is recommended.
+        /// </remarks>
+        public ISerializationSurrogateProvider? DataContractSurrogate { get; set; }
+
```

Comment:
Should this have a different name now as this might get confused as accepting an IDataContractSurrogate?

Reply:
Nope. We discussed that in the API review. The underlying type is different now, but for discoverability we want to keep the property name consistent. And it's one less thing to change in the code when migrating. (Yes, I know... _one less_ because change is necessary anyway.)